LightFramework/src/rendering/scenerenderer.cpp

```diff
@@ -126,29 +122,47 @@ namespace Light {
 
 		Light::Renderer::beginScene(camera, camera.getViewMatrix());
 
-		std::vector<PointLight> lights;
+		std::vector<PointLight> pointLights;
+		std::vector<SpotLight> spotLights;
+		std::vector<DirectionalLight> directionalLights;
 		{
 			auto view = scene->m_registry.view<LightComponent, TransformComponent>();
-			for(auto& entity: view)
+
+			for (auto& entity : view)
 			{
-				auto[light, transform] = view.get(entity);
-				lights.push_back({transform.position, light.m_lightColor});
+
+				auto [light, transform] = view.get(entity);
+				switch (static_cast<int>(light.m_lightType))
+				{
+				case 0:
```

Comment:
Use switch cases with the enum rather than with integers

Reply:
done